pkg/pxc/app/binlogcollector/binlog-collector.go

```diff
@@ -65,6 +66,13 @@ func GetDeployment(cr *api.PerconaXtraDBCluster, initImage string) (appsv1.Deplo
 			labels[key] = value
 		}
 	}
+	var matchLabels map[string]string
+	if existingSelector != nil && !reflect.DeepEqual(existingSelector.MatchLabels, naming.LabelsPITR(cr)) {
+		matchLabels = existingSelector.MatchLabels
+	} else {
+		matchLabels = naming.LabelsPITR(cr)
+	}
+
```

Comment:
We can simplify this logic by doing the following:

```
	matchLabels := naming.LabelsPITR(cr)
	if existingSelector != nil && !reflect.DeepEqual(existingSelector.MatchLabels, matchLabels) {
		matchLabels = existingSelector.MatchLabels
	}
```

Is that correct?

Reply:
yes, it looks better! I like it.